shell.nix

```diff
@@ -1,24 +1,16 @@
-{ nixroot  ? (import <nixpkgs> {})
+{ pkgs  ? (import <nixpkgs> {})
```

Comment:
Thank you for this change. Is this a 2.4 candidate as well?
For my understanding this file was not maintained for a while. 

Would you mind to do the nix changes in a separate PR that can merged independently and hopefully faster? 

How does it compare to the discussion in this stale PR? 
https://github.com/mixxxdj/mixxx/pull/3795

Reply:
I haven't really thought about release as I'm not familiar with how 2.4-2.6 is going and this is my first contribution. I'm really for a modern [Flake](https://nixos.wiki/wiki/Flakes) setup and for the idea of having the flake.nix house [both](https://github.com/mixxxdj/mixxx/pull/4632) the development shell and a package for Mixxx.

As for #3795, I don't think it does anything vital compared to my changes. I put in new dependencies added by the Nixpkgs maintainers and myself.

For licensing, idk what the deal is with that as nixpkgs is licensed MIT and I copied the dependency list from there to mixxx/shell.nix.

I also had to repackage modern/pre-release (main branch before 2.4.0 but after 2.3.6) Mixxx for my computers by modifying nixpkgs' derivation: https://gist.github.com/axelkar/1bd63591332b93ddc9f36698f106c92f